docs/index.asciidoc

```diff
@@ -518,11 +518,8 @@ if enabled, script is in charge of creating non-existent document (scripted upda
   * Value type is <<boolean,boolean>>
   * Default value is `false`
 
-This setting asks Elasticsearch for the list of all cluster nodes and adds them to the hosts list.
-Note: This will return ALL nodes with HTTP enabled (including master nodes!). If you use
-this with master nodes, you probably want to disable HTTP on them by setting
-`http.enabled` to false in their elasticsearch.yml. You can either use the `sniffing` option or
-manually enter multiple Elasticsearch hosts using the `hosts` parameter.
+This setting asks Elasticsearch for the list of all cluster nodes and adds the data serving nodes to the hosts list.
```

Comment:
This documentation should state the difference between 5.x and 2.x, where no such sniffing will occur.

Reply:
sniff_2x_1x() already filters for nodes with the data attribute so there should be no difference after this change with 5.x vs 2.x now as far as filtering out nodes.

         # Only connect to nodes that serve data
         # this will skip connecting to client, tribe, and master only nodes
         # Note that if 'attributes' is NOT set, then that's just a regular node
         # with master + data + client enabled, so we allow that
         attributes = info['attributes']
         next if attributes && attributes['data'] == 'false'

Reply:
Hmm scratch that. If the nodes have the 'data' attribute set then sniffing will only include those nodes, but by default ES 2.x does not include that attribute so sniffing would fallback to checking for http.

    {
      "cluster_name" : "elasticsearch",
      "nodes" : {
        "JS_X6zdyQ2-K9AWBGbdbhw" : {
          "name" : "Mockingbird",
          "transport_address" : "127.0.0.1:9300",
          "host" : "127.0.0.1",
          "ip" : "127.0.0.1",
          "version" : "2.4.6",
          "build" : "5376dca",
          "http_address" : "127.0.0.1:9200",
          "http" : {
            "bound_address" : [ "[::1]:9200", "127.0.0.1:9200" ],
            "publish_address" : "127.0.0.1:9200",
            "max_content_length_in_bytes" : 104857600
          }
        }
      }
    }